### ISSUE TITLE
test: obj_critnib_mt/TEST1: SETUP (all/pmem/debug/helgrind)

### ISSUE THREAD

**Issue by lukaszstolarczuk** [pmem/issues#1005]
## Environment Information 
- PMDK package version(s): 1.5-252-ga8c65dfa7
- OS(es) version(s): Fedora 27 / Ubuntu 18.04
- kernel version(s): 4.15.3 / 4.15.0

## Please provide a reproduction of the bug:
```
./RUNTESTS obj_critnib_mt -s TEST1 -t all
```

## How often bug is revealed:
always
## Actual behavior:
```
obj_critnib_mt/TEST1 failed with Valgrind. See helgrind1.log. Last 20 lines below.
yes: standard output: Broken pipe
obj_critnib_mt/TEST1 helgrind1.log ==12115== Locks held: 1, at address 0x6BBBB60
obj_critnib_mt/TEST1 helgrind1.log ==12115==    at 0x405948: store (critnib.c:171)
obj_critnib_mt/TEST1 helgrind1.log ==12115==    by 0x4063B3: critnib_remove (critnib.c:517)
obj_critnib_mt/TEST1 helgrind1.log ==12115==    by 0x404DEB: thread_write1024 (obj_critnib_mt.c:132)
obj_critnib_mt/TEST1 helgrind1.log ==12115==    by 0x4C369E1: mythread_wrapper (hg_intercepts.c:389)
obj_critnib_mt/TEST1 helgrind1.log ==12115==    by 0x593161A: start_thread (in /usr/lib64/libpthread-2.26.so)
obj_critnib_mt/TEST1 helgrind1.log ==12115==    by 0x5C6098E: clone (in /usr/lib64/libc-2.26.so)
obj_critnib_mt/TEST1 helgrind1.log ==12115==  Address 0x6bbba40 is 0 bytes inside a block of size 336 alloc'd
obj_critnib_mt/TEST1 helgrind1.log ==12115==    at 0x4C30F7B: malloc (vg_replace_malloc.c:299)
obj_critnib_mt/TEST1 helgrind1.log ==12115==    by 0x421A5D: Zalloc (util.c:68)
obj_critnib_mt/TEST1 helgrind1.log ==12115==    by 0x4059C8: critnib_new (critnib.c:218)
obj_critnib_mt/TEST1 helgrind1.log ==12115==    by 0x405184: test (obj_critnib_mt.c:217)
obj_critnib_mt/TEST1 helgrind1.log ==12115==    by 0x405520: main (obj_critnib_mt.c:281)
obj_critnib_mt/TEST1 helgrind1.log ==12115==  Block was alloc'd by thread #1
obj_critnib_mt/TEST1 helgrind1.log ==12115== 
obj_critnib_mt/TEST1 helgrind1.log ==12115== 
obj_critnib_mt/TEST1 helgrind1.log ==12115== For counts of detected and suppressed errors, rerun with: -v
obj_critnib_mt/TEST1 helgrind1.log ==12115== Use --history-level=approx or =none to gain increased speed, at
obj_critnib_mt/TEST1 helgrind1.log ==12115== the cost of reduced accuracy of conflicting-access information
obj_critnib_mt/TEST1 helgrind1.log ==12115== ERROR SUMMARY: 10985 errors from 3 contexts (suppressed: 110507 from 9)
RUNTESTS: stopping: obj_critnib_mt/TEST1 failed, TEST=all FS=pmem BUILD=debug
tput: No value for $TERM and no -T specified
tput: No value for $TERM and no -T specified
1 tests failed:
obj_critnib_mt/TEST1
```
Full log: 
[helgrind1.log](https://github.com/pmem/issues/files/2812253/helgrind1.log)

## Expected behavior:
test passes

**Comment by kilobyte** [comment]
Was also broken under drd — added a test for that.

Mitigated by pmem/pmdk#3644 — not as good as we'd want as it issues a valgrind call inside a tight loop, but that's acceptable for 1.6.  Can improve performance later.

**Comment by kilobyte** [comment]
A better fix, one that doesn't hurt performance, has been merged into 1.6.

**Comment by Greg091** [comment]
Verified on: 1.6-rc2-15-g0a0d0821a
